src/page/menu.rs

```diff
@@ -136,49 +144,126 @@ fn NsfwToggle() -> impl IntoView {
 #[component]
 pub fn Menu() -> impl IntoView {
     let (is_connected, _) = account_connected_reader();
+    let query_map = use_query_map();
+    let show_content_modal = create_rw_signal(false);
+
+    let can_res = authenticated_canisters();
+    let is_authorized = create_local_resource_with_initial_value(
+        move || MockPartialEq(can_res()),
+        move |can_res| async move {
+            let canisters = can_res.0?.ok()?;
+            canisters.is_user_authorized_to_seed_content().await.ok()
+        },
+        Some(Some(false)),
+    );
+
+    create_effect(move |_| {
+        //check whether query param is right if right set the show_modal_content as true.
+        let authorized = is_authorized.get().flatten()?;
+        let query_params = query_map.get();
+        let url = query_params.0.get("text")?;
+        if !url.is_empty() && authorized && is_connected.get() {
+            show_content_modal.set(true);
+        }
+        Some(())
+    });
 
     view! {
-        <div class="min-h-screen w-full flex flex-col text-white pt-2 pb-12 bg-black items-center divide-y divide-white/10">
-            <div class="flex flex-col items-center w-full gap-20 pb-16">
-                <Title justify_center=false>
-                    <div class="flex flex-row justify-between">
-                        <BackButton fallback="/".to_string()/>
-                        <span class="font-bold text-2xl">Menu</span>
-                        <div></div>
-                    </div>
-                </Title>
-                <div class="flex flex-col items-center w-full gap-4">
-                    <div class="flex flex-row justify-center gap-4 items-center px-4">
-                        <ProfileInfo/>
-                    </div>
-                    <Show when=move || !is_connected()>
-                        <div class="w-full px-8 md:w-4/12 xl:w-2/12">
-                            <ConnectLogin/>
-                        </div>
-                        <div class="w-full px-8 text-center text-sm font-sans">
-                            {r#"Your Yral account has been setup. Login with Google to not lose progress."#}
+        <Suspense fallback=move || {
```

Comment:
Use a fine grained suspense.
only YoutubeUpload modal should show the loading spinner
UploadContent button can stay hidden till the resource is completed

Reply:
Done